File: kit-fastapi/tests/fakes/fake_id_generator.py
"""Génère des IDs séquentiels déterministes (gen-1, gen-2, …)."""
from __future__ import annotations


class FakeIdGenerator:
    def __init__(self) -> None:
        self._seq = 0

    def uuid(self) -> str:
        self._seq += 1
        return f"gen-{self._seq}"

File: kit-php/tests/Fakes/FakeIdGenerator.php
<?php

declare(strict_types=1);

namespace TodoApp\Tests\Fakes;

use TodoApp\Domain\Todo\IdGenerator;

/** Génère des IDs séquentiels déterministes (gen-1, gen-2, …). */
final class FakeIdGenerator implements IdGenerator
{
    private int $seq = 0;

    public function uuid(): string
    {
        return 'gen-' . (++$this->seq);
    }
}
